airflow/api_fastapi/core_api/routes/public/pools.py

```diff
@@ -167,15 +168,57 @@ def patch_pool(
 @pools_router.post(
     "/",
     status_code=status.HTTP_201_CREATED,
-    responses=create_openapi_http_exception_doc([status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN]),
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN, status.HTTP_409_CONFLICT]
+    ),
 )
 def post_pool(
     post_body: PoolPostBody,
     session: Annotated[Session, Depends(get_session)],
 ) -> PoolResponse:
     """Create a Pool."""
+    pool = session.scalar(select(Pool).where(Pool.pool == post_body.pool))
+    if pool is not None:
+        raise HTTPException(status.HTTP_409_CONFLICT, f"Pool with name: `{post_body.pool}` already exists")
     pool = Pool(**post_body.model_dump())
 
     session.add(pool)
 
     return PoolResponse.model_validate(pool, from_attributes=True)
```

Comment:
```suggestion
def post_pool(
    body: PoolPostBody,
    session: Annotated[Session, Depends(get_session)],
) -> PoolResponse:
    """Create a Pool."""
    pool = session.scalar(select(Pool).where(Pool.pool == body.pool))
    if pool is not None:
        raise HTTPException(status.HTTP_409_CONFLICT, f"Pool with name: `{body.pool}` already exists")
    pool = Pool(**body.model_dump())

    session.add(pool)

    return PoolResponse.model_validate(pool, from_attributes=True)
```

Refer: https://github.com/apache/airflow/pull/42975#discussion_r1838380546